server/etcdserver/server.go

```diff
@@ -316,6 +320,9 @@ func (bh *backendHooks) OnPreCommitUnsafe(tx backend.BatchTx) {
 		// save bh.confState
 		bh.confStateDirty = false
 	}
+	bh.versionStored.Do(func() {
```

Comment:
I think its good enough place for now... but in general I think that the 'version' update happens for good
when we dump first WAL log Snapshot (not commited the first transaction). 

Having a WAL snapshot guarantees that not only bbolt but also WAL logs (the important for replay part) are written with respect to the newly introduced data. For example change of the WAL snapshot: 
  https://github.com/etcd-io/etcd/blob/76df94a8399eafb48eb8a02d6b1b90868d6b4277/server/wal/walpb/record.proto#L23